### PR TITLE
Add a label to prevent upload

### PR DESCRIPTION
**Pull request by superpoussin22**
I think it could be fien to prevent upload to virus total if apk is not in database , but grab the report if in it

### CODE REVIEW

File: StaticAnalyzer/views/android/VirusTotal.py [hunk 1]
```diff
@@ -28,43 +28,44 @@ def get_report(self, file_hash):
...
         '''
-        try:
-            url = self.base_url + "scan"
-            files = {
-                'file': open(file_path, 'rb')
-            }
-            headers = {
-                "apikey": settings.VT_API_KEY
-            }
+        if settings.VT_UPLOAD:
```
Comment (matandobr) [inline StaticAnalyzer/views/android/VirusTotal.py]: This check should not be inside the upload function, the function should do what it says it does,
move this check into the get_result function

**Comment by matandobr** [comment]
After I run a test I get these errors: 

> [INFO] Analysis is already Done. Fetching data from the DB...
[INFO] VirusTotal: Check for existing report

>[2017-11-03 15:56:22]
[ERROR] in VirusTotal get_report (/Users/matandobr/repos/tmp1/Mobile-Security-Framework-MobSF/StaticAnalyzer/views/android/VirusTotal.py, LINE 32 "json_response = response.json()"): No JSON object could be decoded

>[2017-11-03 15:56:22]
[ERROR] in VirusTotal get_result (/Users/matandobr/repos/tmp1/Mobile-Security-Framework-MobSF/StaticAnalyzer/views/android/VirusTotal.py, LINE 83 "print "[INFO] VirusTotal: " + report['verbose_msg']"): 'NoneType' object has no attribute '__getitem__'
[OK] PDF Report Generated: /PDF/?md5=52c50ae824e329ba8b5b7a0f523efffe&type=ANDZIP

>[INFO] Fetching data from DB for PDF Report Generation (IOS ZIP)

>[INFO] Analysis is already Done. Fetching data from the DB...
[INFO] VirusTotal: Check for existing report

>[2017-11-03 15:56:24]
[ERROR] in VirusTotal get_report (/Users/matandobr/repos/tmp1/Mobile-Security-Framework-MobSF/StaticAnalyzer/views/android/VirusTotal.py, LINE 32 "json_response = response.json()"): No JSON object could be decoded

>[2017-11-03 15:56:24]
[ERROR] in VirusTotal get_result (/Users/matandobr/repos/tmp1/Mobile-Security-Framework-MobSF/StaticAnalyzer/views/android/VirusTotal.py, LINE 83 "print "[INFO] VirusTotal: " + report['verbose_msg']"): 'NoneType' object has no attribute '__getitem__'
[OK] PDF Report Generated: /PDF/?md5=57bb5be0ea44a755ada4a93885c3825e&type=IOSZIP

>[INFO] Fetching data from DB for PDF Report Generation (APPX)
[INFO] VirusTotal: Check for existing report

>[2017-11-03 15:56:26]
[ERROR] in VirusTotal get_report (/Users/matandobr/repos/tmp1/Mobile-Security-Framework-MobSF/StaticAnalyzer/views/android/VirusTotal.py, LINE 32 "json_response = response.json()"): No JSON object could be decoded

>[2017-11-03 15:56:26]
[ERROR] in VirusTotal get_result (/Users/matandobr/repos/tmp1/Mobile-Security-Framework-MobSF/StaticAnalyzer/views/android/VirusTotal.py, LINE 83 "print "[INFO] VirusTotal: " + report['verbose_msg']"): 'NoneType' object has no attribute '__getitem__'

**PLEASE RUN THE UNIT TEST** 
http://127.0.0.1:8000/runtest/

**Comment by superpoussin22** [comment]
@matandor   so you test with an ipa right ?
will take a look all those lines comes from original VirusToatl.py not written by me

**Comment by matandobr** [comment]
It doesn't matter where these errors came from, if you made a modification that result in errors you should fix it

**Comment by ajinabraham** [comment]
@matandobr Have you tested this?

**Comment by matandobr** [comment]
Made another review, still same errors I mentioned before!
**PLEASE RUN THE UNIT TEST!!!**

**Comment by superpoussin22** [comment]
@matandobr last test on my side was clean without error  can you post a log so I can try to understand ?

**Comment by matandobr** [comment]
> MacBook-Pro-2:Mobile-Security-Framework-MobSF matandobr$ python manage.py runserver 
> Performing system checks...
> 
> 
>  __  __       _    ____  _____        ___   ___   ____  
> |  \/  | ___ | |__/ ___||  ___|_   __/ _ \ / _ \ | ___| 
> | |\/| |/ _ \| '_ \___ \| |_  \ \ / / | | | (_) ||___ \ 
> | |  | | (_) | |_) |__) |  _|  \ V /| |_| |\__, | ___) |
> |_|  |_|\___/|_.__/____/|_|     \_/  \___(_) /_(_)____/ 
>                                                         
> 
> 
> 
> Mobile Security Framework v0.9.5.4 Beta
> 
> REST API Key: XXX
> OS: Darwin
> Platform: Darwin-16.6.0-x86_64-i386-64bit
> 
> [INFO] Finding JDK Location in Linux/MAC....
> 
> [INFO] JDK 1.7 or above is available
> 
> [INFO] Checking for Update.
> 
> [INFO] No updates available.
> System check identified no issues (0 silenced).
> November 16, 2017 - 17:51:15
> Django version 1.11.5, using settings 'MobSF.settings'
> Starting development server at http://127.0.0.1:8000/
> Quit the server with CONTROL-C.
> 
> [INFO] Running Static Analyzer Unit test
> [INFO] Running Upload Test
> [INFO] MIME Type: application/zip FILE: android_src.zip
> 
> [INFO] Performing Static Analysis of Android/iOS Source Code
> [OK] Upload OK: android_src.zip
> [INFO] MIME Type: application/vnd.android.package-archive FILE: helloworld.apk
> 
> [INFO] Performing Static Analysis of Android APK
> [OK] Upload OK: helloworld.apk
> [INFO] MIME Type: application/octet-stream FILE: helloworld.ipa
> 
> [INFO] Performing Static Analysis of iOS IPA
> [OK] Upload OK: helloworld.ipa
> [INFO] MIME Type: application/zip FILE: ios_src.zip
> 
> [INFO] Performing Static Analysis of Android/iOS Source Code
> [OK] Upload OK: ios_src.zip
> [INFO] MIME Type: application/octet-stream FILE: Windows-App-Studio-Sample-App-1.8.0.0-Appx4Fun.com.appx
> 
> [INFO] Performing Static Analysis of Windows APP
> [OK] Upload OK: Windows-App-Studio-Sample-App-1.8.0.0-Appx4Fun.com.appx
> [OK] Completed Upload test
> [INFO] Running Static Analysis Test
> [INFO] Starting Analysis on : android_src.zip
> 
> [INFO] Analysis is already Done. Fetching data from the DB...
> [OK] Static Analysis Complete: /StaticAnalyzer/?name=android_src.zip&type=zip&checksum=52c50ae824e329ba8b5b7a0f523efffe
> [INFO] Starting Analysis on : helloworld.apk
> 
> [INFO] Analysis is already Done. Fetching data from the DB...
> [INFO] VirusTotal: Check for existing report
> [INFO] VirusTotal: Scan finished, information embedded
> [OK] Static Analysis Complete: /StaticAnalyzer/?name=helloworld.apk&type=apk&checksum=3a552566097a8de588b8184b059b0158
> [INFO] iOS Static Analysis Started
> 
> [INFO] Analysis is already Done. Fetching data from the DB...
> [INFO] VirusTotal: Check for existing report
> [INFO] VirusTotal: Scan finished, information embedded
> [OK] Static Analysis Complete: /StaticAnalyzer_iOS/?name=helloworld.ipa&type=ipa&checksum=6c23c2970551be15f32bbab0b5db0c71
> [INFO] Starting Analysis on : ios_src.zip
> [INFO] Extracting ZIP
> [INFO] Unzipping
> [INFO] Checking for ZIP Validity and Mode
> [INFO] Redirecting to iOS Source Code Analyzer
> [INFO] iOS Static Analysis Started
> [INFO] iOS Source Code Analysis Started
> [INFO] ZIP Already Extracted
> [INFO] Generating Hashes
> [INFO] Get Files, BIN Plist -> XML, and Normalize
> [INFO] iOS Info.plist Analysis Started
> [INFO] Finding Info.plist in iOS Source
> [INFO] Checking Permissions
> [INFO] Checking for Insecure Connections
> [INFO] Starting iOS Source Code and PLIST Analysis
> [INFO] Performing Malware Check on extracted Domains
> 
> [INFO] Updating Malware Database....
> [INFO] Finished Code Analysis, Email and URL Extraction
> 
> [INFO] Connecting to DB
> 
> [INFO] Saving to Database
> [OK] Static Analysis Complete: /StaticAnalyzer/?name=ios_src.zip&type=zip&checksum=57bb5be0ea44a755ada4a93885c3825e
> [INFO] Windows Static Analysis Started
> [INFO] Windows Binary Analysis Started
> [INFO] Generating Hashes
> [INFO] Extracting APPX
> [INFO] Unzipping
> [INFO] Starting Binary Analysis - XML
> [INFO] Reading AppxManifest
> [INFO] Starting Binary Analysis
> [INFO] Windows VM not configured in settings.py. Skipping Binskim and Binscope.
> 
> [INFO] Connecting to DB
> 
> [INFO] Saving to Database
> [OK] Static Analysis Complete: /StaticAnalyzer_Windows/?name=Windows-App-Studio-Sample-App-1.8.0.0-Appx4Fun.com.appx&type=appx&checksum=8179b557433835827a70510584f3143e
> [OK] Static Analysis test completed
> [INFO] Running PDF Generation Test
> 
> [INFO] Fetching data from DB for PDF Report Generation (Android)
> 
> [INFO] Analysis is already Done. Fetching data from the DB...
> [INFO] VirusTotal: Check for existing report
> [INFO] VirusTotal: Scan finished, information embedded
> [OK] PDF Report Generated: /PDF/?md5=3a552566097a8de588b8184b059b0158&type=APK
> 
> [INFO] Fetching data from DB for PDF Report Generation (IOS IPA)
> 
> [INFO] Analysis is already Done. Fetching data from the DB...
> [INFO] VirusTotal: Check for existing report
> [INFO] VirusTotal: Scan finished, information embedded
> [OK] PDF Report Generated: /PDF/?md5=6c23c2970551be15f32bbab0b5db0c71&type=IPA
> 
> [INFO] Fetching data from DB for PDF Report Generation (Android)
> 
> [INFO] Analysis is already Done. Fetching data from the DB...
> [INFO] VirusTotal: Check for existing report
> 
> [2017-11-16 17:51:38]
> [ERROR] in VirusTotal get_report (/Users/matandobr/repos/Mobile-Security-Framework-MobSF/StaticAnalyzer/views/android/VirusTotal.py, LINE 32 "json_response = response.json()"): No JSON object could be decoded
> 
> [2017-11-16 17:51:38]
> [ERROR] in VirusTotal get_result (/Users/matandobr/repos/Mobile-Security-Framework-MobSF/StaticAnalyzer/views/android/VirusTotal.py, LINE 83 "print "[INFO] VirusTotal: " + report['verbose_msg']"): 'NoneType' object has no attribute '__getitem__'
> [OK] PDF Report Generated: /PDF/?md5=52c50ae824e329ba8b5b7a0f523efffe&type=ANDZIP
> 
> [INFO] Fetching data from DB for PDF Report Generation (IOS ZIP)
> 
> [INFO] Analysis is already Done. Fetching data from the DB...
> [INFO] VirusTotal: Check for existing report
> 
> [2017-11-16 17:51:40]
> [ERROR] in VirusTotal get_report (/Users/matandobr/repos/Mobile-Security-Framework-MobSF/StaticAnalyzer/views/android/VirusTotal.py, LINE 32 "json_response = response.json()"): No JSON object could be decoded
> 
> [2017-11-16 17:51:40]
> [ERROR] in VirusTotal get_result (/Users/matandobr/repos/Mobile-Security-Framework-MobSF/StaticAnalyzer/views/android/VirusTotal.py, LINE 83 "print "[INFO] VirusTotal: " + report['verbose_msg']"): 'NoneType' object has no attribute '__getitem__'
> [OK] PDF Report Generated: /PDF/?md5=57bb5be0ea44a755ada4a93885c3825e&type=IOSZIP
> 
> [INFO] Fetching data from DB for PDF Report Generation (APPX)
> [INFO] VirusTotal: Check for existing report
> 
> [2017-11-16 17:51:42]
> [ERROR] in VirusTotal get_report (/Users/matandobr/repos/Mobile-Security-Framework-MobSF/StaticAnalyzer/views/android/VirusTotal.py, LINE 32 "json_response = response.json()"): No JSON object could be decoded
> 
> [2017-11-16 17:51:42]
> [ERROR] in VirusTotal get_result (/Users/matandobr/repos/Mobile-Security-Framework-MobSF/StaticAnalyzer/views/android/VirusTotal.py, LINE 83 "print "[INFO] VirusTotal: " + report['verbose_msg']"): 'NoneType' object has no attribute '__getitem__'
> [OK] PDF Report Generated: /PDF/?md5=8179b557433835827a70510584f3143e&type=APPX
> [OK] PDF Generation test completed
> [INFO] Running Delete Scan Results test
> [OK] Deleted Scan: 3a552566097a8de588b8184b059b0158
> [OK] Deleted Scan: 6c23c2970551be15f32bbab0b5db0c71
> [OK] Deleted Scan: 52c50ae824e329ba8b5b7a0f523efffe
> [OK] Deleted Scan: 57bb5be0ea44a755ada4a93885c3825e
> [OK] Deleted Scan: 8179b557433835827a70510584f3143e
> [INFO] Delete Scan Results test completed
> 
> 
> [INFO] ALL TESTS COMPLETED!
> [INFO] Test Status: all tests completed
> [16/Nov/2017 17:51:43] "GET /runtest/ HTTP/1.1" 200 47
>

File: MobSF/settings.py [hunk 2]
```diff
@@ -280,13 +280,14 @@
...
     VT_API_KEY = 'XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX'
+    VT_UPLOAD = False
     # Before setting VT_ENABLED to True,
     # Make sure VT_API_KEY is set to your VirusTotal API key
     # register at: https://www.virustotal.com/#/join-us
     # You can get your API KEY from https://www.virustotal.com/en/user/<username>/apikey/
     # VT has a premium features but the free account is just enough for personal use
     # BE AWARE - if you enable VT, in case the file wasn't already uploaded to VirusTotal,
-    # It will be uploaded!
+    # It will be uploaded if you set VT_UPLAOD to True!
```
Comment (ajinabraham) [inline MobSF/settings.py]: Please fix the typo here `VT_UPLOAD` it is.

File: StaticAnalyzer/views/android/VirusTotal.py [hunk 3]
```diff
@@ -79,12 +80,16 @@ def get_result(self, file_path, file_hash):
...
-            return upload_response
+                    return report                  
+            if settings.VT_UPLOAD:
+                print "[INFO] VirusTotal: file upload"
+                upload_response = self.upload_file(file_path)
+                if upload_response:
+                    print "[INFO] VirusTotal: " + upload_response['verbose_msg']
+                return upload_response
+            else:
+                print "[INFO] MobSF: Upload not done VT_UPLOAD set to False in settings.py"
```
Comment (ajinabraham) [inline StaticAnalyzer/views/android/VirusTotal.py]: Can you please reword this to 
```
VirusTotal Scan not performed as file upload is disabled in settings.py. To enable file upload, set VT_UPLOAD to True.
```

File: StaticAnalyzer/views/android/VirusTotal.py [hunk 4]
```diff
@@ -79,12 +80,16 @@ def get_result(self, file_path, file_hash):
...
+                report = {"verbose_msg": "No Report file upload disabled"}
```
Comment (ajinabraham) [inline StaticAnalyzer/views/android/VirusTotal.py]: Reword to `Scan Not performed, VirusTotal file upload disabled in settings.py`

Comment (superpoussin22) [reply StaticAnalyzer/views/android/VirusTotal.py]: done

File: StaticAnalyzer/views/android/VirusTotal.py (hunk 3 above)
Comment (superpoussin22) [reply StaticAnalyzer/views/android/VirusTotal.py]: done

File: MobSF/settings.py (hunk 2 above)
Comment (superpoussin22) [reply MobSF/settings.py]: done

**Comment by ajinabraham** [comment]
Looks Good to me all unit tests are running fine. @matandobr any comments?

**Comment by matandobr** [comment]
Nope, looks good 👍
